packages/exporter-trace-otlp-http/src/OTLPExporterBase.ts

```diff
@@ -102,6 +112,32 @@ export abstract class OTLPExporterBase<
     });
   }
 
+  private _configureTimeout(timeoutMillis: number | undefined): number {
+    return typeof timeoutMillis === 'number'
+      ? timeoutMillis < 0
+        ? this._invalidTimeout(timeoutMillis, DEFAULT_TIMEOUT)
+        : timeoutMillis
+          : this._selectATimeout();
+  }
+
+  private _selectATimeout(): number {
+    const definedTimeout =
+      Number(process.env.OTEL_EXPORTER_OTLP_TRACES_TIMEOUT ||
+      process.env.OTEL_EXPORTER_OTLP_TIMEOUT);
+
+    return definedTimeout
+      ? definedTimeout < 0
+        ? this._invalidTimeout(definedTimeout, DEFAULT_TIMEOUT)
+        : definedTimeout
+          : getEnv().OTEL_EXPORTER_OTLP_TRACES_TIMEOUT;
```

Comment:
I think the idle solution is to have just a single default in environment and use it everywhere.
But I see how it's not so trivial to use in this case.

For reference: [java is throwing an exception](https://github.com/open-telemetry/opentelemetry-java/blob/a047632ad75255ecf8a7ac8d783be842f101651e/exporters/otlp-http/trace/src/main/java/io/opentelemetry/exporter/otlp/http/trace/OtlpHttpSpanExporterBuilder.java#L34) if the timeout is negative, preventing the exporter from being initialized.

Reply:
I see. I went with just providing a warning if the value is negative (and in this case I will add if it's set to 0) because the [spec says to ignore that settings and use the default in that case](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#duration):